### PR TITLE
Module image creation jobs for api-gateway

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- post job creating OCI module image after merging to `mod-dev` branch
- job creating OCI module image for release purposes (triggered by tag creation).

**Related issue(s)**

### CODE REVIEW

File: templates/data/api-gateway-manager-build.yaml [hunk 1]
```diff
@@ -62,22 +62,102 @@ templates:
                     - "image_buildpack-golang" # takes latest golang image
                   local:
                     - job_default
+              - jobConfig: # builds an OCI image with api-gateway module from mod-dev branch
+                  name: post-api-gateway-module-build
+                  annotations:
+                    owner: "goat"
+                    description: "API Gateway Manager template image build on mod-dev branch"
+                  always_run: true
+                  optional: true
```
Comment (kolodziejczak) [inline templates/data/api-gateway-manager-build.yaml:71]: Do we want optional to be set to true here?

File: templates/data/api-gateway-manager-build.yaml [hunk 2]
```diff
@@ -62,22 +62,102 @@ templates:
...
+                    - "image_buildpack-golang" # takes latest golang image
+                  local:
+                    - job_default
+              - jobConfig: # builds an OCI image with api-gateway module from release branch
+                  name: rel-api-gateway-module-build
+                  annotations:
+                    owner: "goat"
+                    description: "API Gateway Manager template image build for release"
+                  always_run: true
+                  optional: true
```
Comment (kolodziejczak) [inline templates/data/api-gateway-manager-build.yaml:100]: Do we want optional to be set to true here?

File: templates/data/api-gateway-manager-build.yaml [hunk 3]
```diff
@@ -62,22 +62,102 @@ templates:
                     - "image_buildpack-golang" # takes latest golang image
                   local:
                     - job_default
+              - jobConfig: # builds an OCI image with api-gateway module from mod-dev branch
+                  name: post-api-gateway-module-build
```
Comment (kolodziejczak) [inline templates/data/api-gateway-manager-build.yaml:66]: Would be good to set slack notifications for this job if it fails

File: templates/data/api-gateway-manager-build.yaml [hunk 4]
```diff
@@ -62,22 +62,102 @@ templates:
...
+                    - ^mod-dev$
+                inheritedConfigs:
+                  global:
+                    - "jobConfig_default"
+                    - "jobConfig_postsubmit"
+                    - "image_buildpack-golang" # takes latest golang image
+                  local:
+                    - job_default
+              - jobConfig: # builds an OCI image with api-gateway module from release branch
+                  name: rel-api-gateway-module-build
```
Comment (kolodziejczak) [inline templates/data/api-gateway-manager-build.yaml:95]: Would be good to set slack notifications for this job if it fails

**Review by kolodziejczak — CHANGES_REQUESTED** [review]

File: templates/data/api-gateway-manager-build.yaml (hunk 1 above)
Comment (jaroslaw-pieszka) [reply templates/data/api-gateway-manager-build.yaml:71]: In the next pushed version.

File: templates/data/api-gateway-manager-build.yaml (hunk 2 above)
Comment (jaroslaw-pieszka) [reply templates/data/api-gateway-manager-build.yaml:100]: In the next pushed version.

File: templates/data/api-gateway-manager-build.yaml (hunk 3 above)
Comment (jaroslaw-pieszka) [reply templates/data/api-gateway-manager-build.yaml:66]: In the next commit.

File: templates/data/api-gateway-manager-build.yaml (hunk 4 above)
Comment (jaroslaw-pieszka) [reply templates/data/api-gateway-manager-build.yaml:95]: In the next commit.

**Review by kolodziejczak — APPROVED** [review]

**Review by Sawthis — APPROVED** [review]

**Pull request merged**